### PR TITLE
Remove hardcoded pwd from SqlClient test

### PR DESCRIPTION
**Pull request by pjanotti**
Removing hardcoded password from integration test. The password is now passed as a parameter to the test application, it is not using a secret store (the ideal) but a guid seems good enough to avoid the hard coded password.

### CODE REVIEW

**Review by pellared — COMMENTED** [review]
To be honest, I do not see any issue in having hardcoded creds in test applications.
This will only make a little trouble when one would like to interactively log in to the SQL Server (e.g. when debugging).

File: test/integration-tests/IntegrationTests.SqlClient/SqlClientCollection.cs [hunk 1]
```diff
@@ -33,7 +34,6 @@ public class SqlClientFixture : IAsyncLifetime
     {
         private const int DatabasePort = 1433;
         private const string DatabaseImage = "mcr.microsoft.com/mssql/server:2019-CU15-ubuntu-20.04";
-        private const string DatabasePassword = "@someThingComplicated1234";
```
Comment (nrcventura) [inline test/integration-tests/IntegrationTests.SqlClient/SqlClientCollection.cs:36]: Isn't this password still hardcoded in one of the yml files? Perhaps one of the other changes in this PR is overriding that setting, but it's not clear to me?

Comment (pjanotti) [reply test/integration-tests/IntegrationTests.SqlClient/SqlClientCollection.cs:36]: My VS Code doesn't find `@someThingComplicated1234` in any other place.

Comment (nrcventura) [reply test/integration-tests/IntegrationTests.SqlClient/SqlClientCollection.cs:36]: It looks like I was remembering a change in an [older commit](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/669/commits/80226cdef371e8fc7a00c9bab9f8235baaf0cd38) that was later removed.

**Review by pellared — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**